internal/exec/stages/fetch_offline/fetch-offline.go

```diff
@@ -84,13 +84,26 @@ func configNeedsNet(cfg *types.Config) (bool, error) {
 func configNeedsNetRecurse(v reflect.Value) (bool, error) {
 	t := v.Type()
 	k := t.Kind()
+
 	switch {
 	case cfgutil.IsPrimitive(k):
 		return false, nil
 	case t == reflect.TypeOf(types.Resource{}):
 		return sourceNeedsNet(v.Interface().(types.Resource))
 	case t == reflect.TypeOf(types.Tang{}):
 		return true, nil
+	case t == reflect.TypeOf(types.ClevisCustom{}):
+		for i := 0; i < v.NumField(); i += 1 {
+			if v.Type().Field(i).Name == "NeedsNetwork" {
+				if v.Field(i).Kind() == reflect.Ptr {
+					fv := v.Field(i).Elem()
+					if fv.IsValid() {
+						return fv.Bool(), nil
+					}
+				}
+			}
```

Comment:
Instead of this, we should be able to cast to `types.ClevisCustom` here since we know that's its type. That should simplify things greatly.

Reply:
Wow, yeah that makes sense!